src/main/src/main/java/org/geoserver/config/GeoServerDataDirectory.java

```diff
@@ -1078,6 +1079,19 @@ else if ( l.getResource() instanceof WMSLayerInfo ) {
         return r;
     }
     
+    /**
+     * Set the the root directory to look for WKTLib files as the styles directory
+     */
+    public void setWKTLibRoot() {
+        WKTMarkFactory wp = new WKTMarkFactory();
+        URL styleUrl = null;
+        try {
+            styleUrl = new URL("file:" + root().toString() + "/" + STYLE_DIR);
```

Comment:
This should use:
```
resourceLoader.get(STYLE_DIR).file().toURL();
```

You may want to add some checks (eg. file exists, etc) before calling `toURL()`.